components/kyma-environment-broker/internal/broker/instance_get.go

```diff
@@ -31,7 +36,19 @@ func (b *GetInstanceEndpoint) GetInstance(ctx context.Context, instanceID string
 
 	inst, err := b.instancesStorage.GetByID(instanceID)
 	if err != nil {
-		return domain.GetInstanceDetailsSpec{}, errors.Wrapf(err, "while getting instance from storage")
+		statusCode := http.StatusNotFound
+		if !dberr.IsNotFound(err) {
+			statusCode = http.StatusInternalServerError
+		}
+		return domain.GetInstanceDetailsSpec{}, apiresponses.NewFailureResponse(err, statusCode, fmt.Sprintf("failed to get instanceID %s", instanceID))
+	}
+
+	// check if provisioning still in progress
+	if op, err := b.operationsStorage.GetProvisioningOperationByInstanceID(instanceID); err != nil {
+		return domain.GetInstanceDetailsSpec{}, apiresponses.NewFailureResponse(err, http.StatusNotFound, fmt.Sprintf("failed to get operation for instanceID %s", instanceID))
+	} else if op.State == domain.InProgress || op.State == domain.Failed {
+		err = fmt.Errorf("provisioning of instanceID %s %s", instanceID, op.State)
+		return domain.GetInstanceDetailsSpec{}, apiresponses.NewFailureResponse(err, http.StatusNotFound, err.Error())
```

Comment:
All errors this can return are
* when instance storage returns error different from `NotFound`
```
503 failed to get instanceID d3d5dca4-5dc8-44ee-a825-755c2a3fb839
```
* when instance storage returns `NotFound` error
```
404 failed to get instanceID d3d5dca4-5dc8-44ee-a825-755c2a3fb839
```
* when instance storage returns `OK` but operation storage returns error
```
404 failed to get operation for instanceID d3d5dca4-5dc8-44ee-a825-755c2a3fb839
```
* when instance storage returns `OK` but operation storage returns operation that is `InProgress`
```
404 provisioning of instanceID d3d5dca4-5dc8-44ee-a825-755c2a3fb839 in progress
```
* when instance storage returns `OK` but operation storage returns operation that is `Failed`
```
404 provisioning of instanceID d3d5dca4-5dc8-44ee-a825-755c2a3fb839 failed
```
In case of operation errors, users should be capable to fetch more details from the operation endpoint